### ISSUE TITLE
Clicking outside combo box shouldn't select item

### ISSUE THREAD

**Issue by ronnyroeller** [vaadin/vaadin-combo-box#819]
### Description
1. Open the dropdown of the combobox
2. Highlight one of the item in the combo box (e.g. by using up/down cursor)
3. Click outside the dropdown to close it

### Expected outcome
The highlighted item is not selected (e.g. the selection is cancelled similar to pressing ESC).
I'd expect the value to be only selected if the user clicks the item or presses Enter.

In case the current behavior is by design (e.g. clicking outside should mean "item selected"): It would be helpful to expose in the `selected-item-changed` event why the item was selected. This would allow the embedding application to ignore events that were caused by clicking outside the combobox.

### Actual outcome
The highlighted item is selected, i.e. `selected-item-changed` event is fired.

### Live Demo
https://capricious-bug.glitch.me (standard demo)

### Steps to reproduce
1. Highlight one of the values in the combo box (e.g. by using cursor)
2. Click outside the dropdown to close it

### Workaround

This workaround prevents at least the auto-selection of matching text:
```javascript
comboBox.dataProvider = (params: any, callback: any) => {
  // Vaadin's combobox has logic that assumes a suggestion should be focused when the user input matches
  // the label of the suggestion. If the dropdown is closed with a focused item, then that item will be
  // reported as "selected", and we will not be able to determine whether that closing was caused by a
  // click inside ("accept") or outside ("cancel") of the box.
  // So we assign a label that doesn't match the user input but that looks "just like it", preventing the
  // automatic focusing/selection from happening. The user can still focus the suggestion explicitly using
  // the keyboard arrow keys or by pointing to it with the mouse.
  const suggestions = computeSuggestions(params.filter)
    .map(suggestion => {
      const label = suggestion.name ? `${suggestion.name}\u00A0` : suggestion.name;
      return Object.assign({label}, suggestion);
  });
  callback(suggestions, suggestions.length);
};
```

**Comment by HJK181** [comment]
@ronnyroeller The workaround does not work for me because the `vaadin-combo-box-mixin` compares the complete item in the `_onClosed` method:
```
...
 if (this.selectedItem !== focusedItem) {
          this.selectedItem = focusedItem;
        }
...
```
Tested with version `4.2.9` of `vadin-combo-box`. Do you have any clue why it's working for you?

**Comment by HJK181** [comment]
This workaround is not sufficient for us to use it in production, as one is still able to navigate with the keyboard to an item, which will select it if you click outside.

Is there a chance that this issue will be fixed, as it's a show stopper for us. What I'm really missing is just an event which fires on a click of the `vaadin-combo-box-item`, which is actually there `selection-changed` but stopped from propagation in the `vaadin-combo-box-mixin`.

**Comment by jouni** [comment]
> In case the current behavior is by design

Yes, it is by design. The current behavior is documented by the flow chart in https://github.com/vaadin/vaadin-combo-box/issues/726

One of the reasons for it is that the ComboBox component in Vaadin Framework 8 worked this way, and existing users expected it to remain the same.

Another reason is that it makes it a little bit faster to enter data, when the user can press Tab to move to the next field (vs first press Enter, then Tab). 

We can can debate whether blurring the field by pressing Tab or clicking outside it are considered as the same user intent. So far we’ve kept them the same.

**Comment by jouni** [comment]
Issue potential issue with the current behavior is that the highlight style of the matching item (2px blue focus outline) is not prominent enough for the user to notice that it will be selected when the field is blurred.
